### PR TITLE
[zh] Resync secret

### PR DESCRIPTION
**Pull request by tengqm**
This one is big. It is a rework because the upstream Secret concept was almost rewritten.

### CODE REVIEW

File: content/zh/docs/concepts/configuration/secret.md [hunk 1]
```diff
@@ -72,1921 +75,2113 @@ In order to safely use Secrets, take at least the following steps:
...
 
 <!--
-The output looks like:
+#### Constraints on Secret names and data {#restriction-names-data}
+
+The name of a Secret object must be a valid
+[DNS subdomain name](/docs/concepts/overview/working-with-objects/names#dns-subdomain-names).
 -->
-输出类似于
+#### 对 Secret 名称与数据的约束a {#restriction-names-data}
```
Comment (mengjiao-liu) [inline content/zh/docs/concepts/configuration/secret.md:204]:
```suggestion
#### 对 Secret 名称与数据的约束 {#restriction-names-data}
```

File: content/zh/docs/concepts/configuration/secret.md [hunk 2]
```diff
@@ -72,1921 +75,2113 @@ In order to safely use Secrets, take at least the following steps:
...
+The tokens obtained using this method have bounded lifetimes, and are automatically
+invalidated when the Pod they are mounted into is deleted.
+-->
+Kubernetes v1.22 版本之前都会自动创建用来访问 Kubernetes API 的凭证。
+这一老的机制是基于创建可被挂载到 Pod 中的令牌 Secret 来实现的。
+在最近的版本中，包括 Kubernetes v{{< skew currentVersion >}} 中，API 凭据是直接通过
+[TokenRequest](/docs/reference/kubernetes-api/authentication-resources/token-request-v1/)
+API 来获得的，这以凭据会使用[投射卷](/zh/docs/reference/access-authn-authz/service-accounts-admin/#bound-service-account-token-volume)
+挂载到 Pod 中。使用这种方式获得的令牌有确定的生命期，并且在挂载它们的 Pod
+被删除时自动废弃。
```
Comment (mengjiao-liu) [inline content/zh/docs/concepts/configuration/secret.md:439]: “废弃” 改为“失效” 感觉更准确一点

File: content/zh/docs/concepts/configuration/secret.md [hunk 3]
```diff
@@ -72,1921 +75,2113 @@ In order to safely use Secrets, take at least the following steps:
...
-考虑一个需要处理 HTTP 请求、执行一些复杂的业务逻辑，然后使用 HMAC 签署一些消息的应用。
-因为应用程序逻辑复杂，服务器中可能会存在一个未被注意的远程文件读取漏洞，
-可能会将私钥暴露给攻击者。
+Kubernetes 允许你将特定的 Secret（和 ConfigMap）标记为 **不可更改（Immutable）**。
+禁止更改现有 Secret 的数据有下列好处：
+
+- 防止意外（或非预期的）更新导致应用程序中断
+- （对于大量使用 Secret 的集群而言，至少数万个不同的 Secret 供 Pod 挂载），
+  通过将 Secret 标记为不可变，可以极大降低 kube-apiserver 的负载，提升集群性能。
+  kubelet 不需要监视哪些被标记为不可更改的 Secret。
```
Comment (mengjiao-liu) [inline content/zh/docs/concepts/configuration/secret.md:1988]:
```suggestion
  kubelet 不需要监视那些被标记为不可更改的 Secret。
```

File: content/zh/docs/concepts/configuration/secret.md [hunk 4]
```diff
@@ -72,1921 +75,2113 @@ In order to safely use Secrets, take at least the following steps:
...
-### 客户端使用 Secret API
+Although ConfigMap and Secret work similarly, Kubernetes applies some additional
+protection for Secret objects.
+-->
+## Secret 的信息安全问题
 
-当部署与 Secret API 交互的应用程序时，应使用
-[鉴权策略](/zh/docs/reference/access-authn-authz/authorization/)，
-例如 [RBAC](/zh/docs/reference/access-authn-authz/rbac/)，来限制访问。
+尽管 ConfigMap 和 Secret 的工作方式类似，Kubernetes 对 Secret 有一些额外的保护。
```
Comment (mengjiao-liu) [inline content/zh/docs/concepts/configuration/secret.md:2037]:
```suggestion
尽管 ConfigMap 和 Secret 的工作方式类似，但 Kubernetes 对 Secret 有一些额外的保护。
```

Comment (tengqm) [reply content/zh/docs/concepts/configuration/secret.md:2037]: 中文里面，“尽管 ...，但 ...” 这种搭配很奇怪

File: content/zh/docs/concepts/configuration/secret.md [hunk 5]
```diff
@@ -72,1921 +75,2112 @@ In order to safely use Secrets, take at least the following steps:
...
+- [create Secret using kustomize](/docs/tasks/configmap-secret/managing-secret-using-kustomize/)
 -->
-### Opaque Secret
+## 使用 Secret  {#working-with-secrets}
 
-当 Secret 配置文件中未作显式设定时，默认的 Secret 类型是 `Opaque`。
-当你使用 `kubectl` 来创建一个 Secret 时，你会使用 `generic` 子命令来标明
-要创建的是一个 `Opaque` 类型 Secret。
-例如，下面的命令会创建一个空的 `Opaque` 类型 Secret 对象：
+### 创建 Secret  {#creating-a-secret}」
```
Comment (SataQiu) [inline content/zh/docs/concepts/configuration/secret.md:192]:
```suggestion
### 创建 Secret  {#creating-a-secret}
```

File: content/zh/docs/concepts/configuration/secret.md [hunk 6]
```diff
@@ -72,1921 +75,2112 @@ In order to safely use Secrets, take at least the following steps:
...
+Individual secrets are limited to 1MiB in size. This is to discourage creation
+of very large secrets that could exhaust the API server and kubelet memory.
+However, creation of many smaller secrets could also exhaust memory. You can
+use a [resource quota](/docs/concepts/policy/resource-quotas/) to limit the
+number of Secrets (or other resources) in a namespace.
+-->
+#### 尺寸限制   {#restriction-data-size}
 
-The following example configuration declares a service account token Secret:
+每个 Secret 的尺寸最多为 1MiB。施加这一限制时为了避免用户创建非常大的 Secret，
```
Comment (SataQiu) [inline content/zh/docs/concepts/configuration/secret.md:243]:
```suggestion
每个 Secret 的尺寸最多为 1MiB。施加这一限制是为了避免用户创建非常大的 Secret，
```

File: content/zh/docs/concepts/configuration/secret.md [hunk 7]
```diff
@@ -72,1921 +75,2112 @@ In order to safely use Secrets, take at least the following steps:
...
+are obtained directly by using the [TokenRequest](/docs/reference/kubernetes-api/authentication-resources/token-request-v1/) API,
+and are mounted into Pods using a [projected volume](/docs/reference/access-authn-authz/service-accounts-admin/#bound-service-account-token-volume).
+The tokens obtained using this method have bounded lifetimes, and are automatically
+invalidated when the Pod they are mounted into is deleted.
+-->
+Kubernetes v1.22 版本之前都会自动创建用来访问 Kubernetes API 的凭证。
+这一老的机制是基于创建可被挂载到 Pod 中的令牌 Secret 来实现的。
+在最近的版本中，包括 Kubernetes v{{< skew currentVersion >}} 中，API 凭据是直接通过
+[TokenRequest](/docs/reference/kubernetes-api/authentication-resources/token-request-v1/)
+API 来获得的，这以凭据会使用[投射卷](/zh/docs/reference/access-authn-authz/service-accounts-admin/#bound-service-account-token-volume)
```
Comment (SataQiu) [inline content/zh/docs/concepts/configuration/secret.md:437]:
```suggestion
API 来获得的，这一凭据会使用[投射卷](/zh/docs/reference/access-authn-authz/service-accounts-admin/#bound-service-account-token-volume)
```

File: content/zh/docs/concepts/configuration/secret.md [hunk 8]
```diff
@@ -72,1921 +75,2112 @@ In order to safely use Secrets, take at least the following steps:
...
+- `mysecret` 中的键 `username` 会出现在容器中的路径为 `/etc/foo/my-group/my-username`，
+  而不是 `/etc/foo/username`。
+- Secret 对象的 `password` 键不会被投射。
 
-- `username`: 用于身份认证的用户名；
-- `password`: 用于身份认证的密码或令牌。
+如果使用了 `.spec.volumes[].secret.items`，则只有 `items` 中指定了的主键会被投射。
+如果要使用 Secret 中的所有主键，则需要将它们全部枚举到 `items` 字段中。
+
+如果你显式地列举了主键，则所列举的主键都不需在对应的 Secret 中存在。
```
Comment (SataQiu) [inline content/zh/docs/concepts/configuration/secret.md:508]:
```suggestion
如果你显式地列举了主键，则所列举的主键都必须在对应的 Secret 中存在。
```

File: content/zh/docs/concepts/configuration/secret.md (hunk 5 above)
Comment (tengqm) [reply content/zh/docs/concepts/configuration/secret.md:192]: Whoops, I really dislike the Mac IME ...

**Comment by tengqm** [comment]
Thanks @SataQiu, all fixed.

**Review by SataQiu — APPROVED** [review]
thanks @tengqm
/lgtm
/approve

**Pull request merged**